### PR TITLE
chore(cd): add semantic-release workflow

### PR DESCRIPTION
**Pull request by hongaar**
Automatically releases each commit to the master branch using formalized commit message convention to document changes in the codebase.

By default, it uses the [Angular Commit Message Conventions](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines), but it can be changed.

To help with the release process, I suggest adding this: https://github.com/apps/semantic-pull-requests

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 12
+      - env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+        run: npx semantic-release
```
Comment (AdrieanKhisbe) [inline .github/workflows/release.yml:20]: Would prefer have it as devDeps, exposed as script. And then call as `npm run release`

**Review by AdrieanKhisbe — COMMENTED** [review]
If I get it right, for each New commit on master, a New version IS created by semantic release. 

One question though, how does it select thé version *[pre]{patch,minor,major}* ?

**Comment by hongaar** [comment]
> One question though, how does it select thé version *[pre]{patch,minor,major}* ?

Great question. It works by looking at the commits and determines the next version from the type prefixes with this plugin: https://github.com/semantic-release/commit-analyzer

**Comment by AdrieanKhisbe** [comment]
@hongaar Thanks for the link, will try to have a look.

Somethings tell me that our release scheme until v2.0 might not fit standard workflow.
We will have probably find a way to make it stick to the premajor alphas

**Comment by Ore4444** [comment]
commit-analyser can easily fit this workflow.
Just set all "release" values to "premajor".
https://github.com/semantic-release/commit-analyzer/blob/master/lib/default-release-types.js

**Comment by hongaar** [comment]
Not too important right now, closing in favour of more pressing issues.

**Pull request closed without merging**